haystack/preview/components/reader.py

```diff
@@ -0,0 +1,239 @@
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+import os
+from haystack.preview import component, Document, Answer, Pipeline
+from haystack.lazy_imports import LazyImport
+from haystack.nodes.prompt.invocation_layer.utils import get_task
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as torch_and_transformers_import:
+    from transformers import AutoModelForQuestionAnswering, AutoTokenizer
+    from tokenizers import Encoding
+    import torch
+
+
+@component
+class ExtractiveReader:
+    def __init__(
+        self,
+        model: Union[Path, str],
+        device: Optional[str] = None,
+        max_seq_length: int = 384,
+        top_k: int = 10,
```

Comment:
How hard to you think it would be to add support for a `top_p` parameter as well? If it's near trivial I'd love to have that, but if it requires some thinking we can implement it separately.

Reply:
I have added. However, it will not work perfectly for very high values of `top_p` (but you also wouldn't want those at they could return 1000s of answers.
This also shows an advantage of this way of calculating probabilities because `top_p` wouldn't make sense in `FARMReader` as the scores do not add up to 1.